include/osquery/hash.h

```diff
@@ -0,0 +1,25 @@
+/*
+ *  Copyright (c) 2014, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant 
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#ifdef __APPLE__
+  #import <CommonCrypto/CommonDigest.h>
+#else
+  #include <openssl/sha.h>
+  #include <openssl/md5.h>
+#endif
+
+#include <string>
+#include <iomanip>
+#include <sstream>
```

Comment:
If you're only using these for printing the hex digest of the computed hash bytes then move them as close to the implementation as possible (aka into the source/object implementation code). Since we're using a variety of languages (C/C++/ObjC++/ObjC) we must be conservative about where we include code.